### PR TITLE
Emit java method name to IDL/Snapshot to enable generating java stub …

### PR DESCRIPTION
**Pull request by mchen07**
…from IDL

### CODE REVIEW

File: restli-tools/src/test/resources/idls/twitter-statusPromises.restspec.json [hunk 1]
```diff
@@ -20,42 +20,50 @@
...
       "name" : "user_timeline",
+      "javaMethodName" : "getUserTimeline",
       "javaMethodType" : "PROMISE",
       "javaReturnsResultWrapper" : false,
       "doc" : "Gets the status timeline for a given user",
       "pagingSupported" : true
     } ],
     "batchFinders" : [ {
       "name" : "batchFinderByAction",
+      "javaMethodName" : "batchFindStatuses",
```
Comment (junchuanwang) [inline restli-tools/src/test/resources/idls/twitter-statusPromises.restspec.json:66]: This name differs from pojo (StatusCollectionResource.java) method name as well as BatchFinder value, is this expected?
```
  @BatchFinder(value="batchFinderByAction",  batchParam="action")
  public BatchFinderResult<Status, Status, EmptyRecord> batchFinderByAction(
      @QueryParam("action") Status[]  actionNames,
      @QueryParam("q") @Optional String qParam
  )
  {
    return null;
  }
```

Comment (mchen07) [reply restli-tools/src/test/resources/idls/twitter-statusPromises.restspec.json:66]: This IDL is generated from `StatusPromiseResource`, which has a method like this:

```
  @BatchFinder(value="batchFinderByAction",  batchParam="criteria")
  public Promise<BatchFinderResult<Status, Status, User>> batchFindStatuses(@QueryParam("criteria") Status[] criteria)
  {
    return null;
  }
  ```

Comment (junchuanwang) [reply restli-tools/src/test/resources/idls/twitter-statusPromises.restspec.json:66]: thanks. Didn't refresh my repo so I was looking at the wrong file.

**Review by junchuanwang — APPROVED** [review]
looks good!

**Pull request merged**